gazebo_ros/scripts/spawn_entity.py

```diff
@@ -0,0 +1,319 @@
+        initial_pose.position.y = float(self.args.y)
+        initial_pose.position.z = float(self.args.z)
+
+        q = quaternion_from_euler(self.args.R, self.args.P, self.args.Y)
+        initial_pose.orientation.w = q[0]
+        initial_pose.orientation.x = q[1]
+        initial_pose.orientation.y = q[2]
+        initial_pose.orientation.z = q[3]
+
+        success = self._spawn_entity(entity_xml, initial_pose)
+        if not success:
+            self.get_logger().error('Spawn service failed. Exiting.')
+            return 1
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # Apply joint positions if any specified
+        # if len(self.args.joints) != 0:
+        #     joint_names = [joint[0] for joint in self.args.joints]
+        #     joint_positions = [joint[1] for joint in self.args.joints]
+        #     success = _set_model_configuration(joint_names, joint_positions)
+        #     if not success:
+        #         self.get_logger().error('SetModelConfiguration service failed. Exiting.')
+        #         return 1
+
+        # Unpause physics if user requested
+        if self.args.unpause:
+            client = self.create_client(Empty, '%s/unpause_physics' % self.args.gazebo_namespace)
+            client.wait_for_service(timeout_sec=5.0)
+            self.get_logger().info(
+                'Calling service %s/unpause_physics' % self.args.gazebo_namespace)
+            client.call_async(Empty.Request())
+
+        # TODO(shivesh): Wait for https://github.com/ros2/rclpy/issues/244
+        # If bond enabled, setup shutdown callback and wait for shutdown
+        # if self.args.bond:
+        #     rospy.on_shutdown(self._delete_entity)
+        #     rospy.loginfo('Waiting for shutdown to delete entity {}'.format(self.args.entity))
+        #     rospy.spin()
+
+        return 0
+
+    def _spawn_entity(self, entity_xml, initial_pose):
+        self.get_logger().info('Waiting for service %s/spawn_entity' % self.args.gazebo_namespace)
+        client = self.create_client(SpawnEntity, '%s/spawn_entity' % self.args.gazebo_namespace)
+        client.wait_for_service(timeout_sec=5.0)
+        req = SpawnEntity.Request()
+        req.name = self.args.entity
+        req.xml = str(entity_xml, 'utf-8')
+        req.robot_namespace = self.args.robot_namespace
+        req.initial_pose = initial_pose
+        req.reference_frame = self.args.reference_frame
+        self.get_logger().info('Calling service %s/spawn_entity' % self.args.gazebo_namespace)
+        srv_call = client.call_async(req)
+        while rclpy.ok():
+            if srv_call.done():
+                self.get_logger().info('Spawn status: %s' % srv_call.result().status_message)
+                break
+            rclpy.spin_once(self)
+        return srv_call.result().success
+
+    # TODO(shivesh): Wait for https://github.com/ros2/rclpy/issues/244
```

Comment:
That issue is closed, so I wonder if there's an alternative way of hooking to shutdown, or if there won't ever be a good way of implementing the bond functionality.

Reply:
I could not figure out any alternative. Should we remove this functionality then?

Reply:
I left a comment on the issue: https://github.com/ros2/rclpy/issues/244#issuecomment-511832721 Let's see if someone comes back with an alternative.

For now, we can proceed with this PR leaving the code commented out.